ui/lib/core/addon/helpers/has-capability.ts

```diff
@@ -9,46 +9,62 @@
  * provide the full map of capabilities along with the id of the item and capabilities to be evaluated (read, update etc)
  * multiple capabilities will return true if any of the capabilities are true by default
  * to check if all provided capability types are true, set the all argument to true
- * usage example: {{has-capability this.capabilities "delete" "update" id=item.id all=true}}
+ * usage example: {{has-capability this.capabilities "delete" "update" pathKey="customMessages" params=message all=true}}
  */
 
-import { helper as buildHelper } from '@ember/component/helper';
+import Helper from '@ember/component/helper';
+import { service } from '@ember/service';
 import { capitalize } from '@ember/string';
 
 import type { Capabilities, CapabilitiesMap, CapabilityTypes } from 'vault/app-types';
+import type CapabilitiesService from 'vault/services/capabilities';
+import type { PATH_MAP } from 'vault/utils/constants/capabilities';
 
-type NamedArgs = {
-  id: string;
-  all?: boolean;
-};
-
-export function hasCapability(
-  capabilitiesMap?: CapabilitiesMap,
-  types?: CapabilityTypes[],
-  id?: string,
-  all?: boolean
-) {
-  if (capabilitiesMap && Array.isArray(types)) {
-    const path = Object.keys(capabilitiesMap).find((key) => (id ? key.includes(id) : false));
-
-    if (path) {
-      const capabilities = capabilitiesMap[path];
-
-      if (capabilities) {
-        const method = all ? 'every' : 'some';
-
-        return types[method]((type) => {
-          const key = `can${capitalize(type)}` as keyof Capabilities;
-          // default to true if type provided is not valid - edit rather than update for example
-          return !(key in capabilities) ? true : capabilities[key];
-        });
+export default class HasCapabilityHelper extends Helper {
+  @service declare readonly capabilities: CapabilitiesService;
+
+  compute<T>(
+    [capabilitiesMap, ...types]: [CapabilitiesMap, ...CapabilityTypes[]],
+    { pathKey, params, all = false }: { pathKey: keyof typeof PATH_MAP; params?: T; all?: boolean }
+  ) {
+    return () => {
+      // since the default is to return true we need to validate the inputs here more thoroughly
+      // this is to help devs from thinking that the checks are working properly when the capability lookup is actually failing
+      if (!capabilitiesMap) {
+        throw new Error('Capabilities map is required as the first positional arg');
+      }
+      if (!types || !types.length) {
+        throw new Error('Capability types are required as positional args');
```

Comment:
```suggestion
        throw new Error('At least one capability type is required as a positional argument.');
```

Reply:
I made a change to the file before seeing this suggestion but manually committed it.